magicctapipe/scripts/lst1_magic/README.md

```diff
@@ -0,0 +1,226 @@
+# Scripts for MAGIC+LST analysis
+
+This folder contains the scripts to perform MAGIC+LST analysis in a semi-automatic way.
+
+Each script can be called from the command line from anywhere in your system. Please run them with `-h` option for the first time to check what are the options available.
+
+
+## Overview
+
+
+MAGIC+LST analysis starts from MAGIC calibrated data (\_Y\_ files), LST  data level 1 (DL1) data and SimTelArray DL0 data, and our goal is to achieve data level 3 (DL3).
+
+Behind the scenes, the semi-automatic scripts will run:
+- `magic_calib_to_dl1` on real MAGIC data, to convert them into DL1 format.
+- `merge_hdf_files.py` on MAGIC data to merge subruns and/or runs together.
+- `lst1_magic_event_coincidence.py` to find coincident events between MAGIC and LST-1, starting from DL1 data.
+- `lst1_magic_stereo_reco.py` to add stereo parameters to the DL1 data.
+- `lst1_magic_train_rfs.py` to train the RFs (energy, direction, classification) on train gamma MCs and protons.
+- `lst1_magic_dl1_stereo_to_dl2.py` to apply the RFs to stereo DL1 data (real and test MCs) and produce DL2 data.
+- `lst1_magic_create_irf.py` to create the IRF.
+- `lst1_magic_dl2_to_dl3.py` to create DL3 files, and `create_dl3_index_files.py` to create DL3 HDU and index files.
+
+From DL3 on, the analysis is done with gammapy.
+
+## Installation
+
+1) The very first step to reduce MAGIC-LST data is to have remote access/credentials to the IT Container, so provide one. Once you have it, the connection steps are the following:
+
+Authorized institute server (Client) &rarr;  ssh connection to CTALaPalma &rarr; ssh connection to cp01/02.
+
+2) Once connected to the IT Container, install magic-cta-pipe (e.g. in your home directory in the IT Container) with the following commands (if you have mamba installed, we recommend you to use it instead of conda. The installation process will be much faster.):
+
+```
+git clone -b Torino_auto_MCP https://github.com/cta-observatory/magic-cta-pipe.git
+cd magic-cta-pipe
+conda env create -n magic-lst -f environment.yml
+conda activate magic-lst
+pip install .
+```
+
+## Analysis
+
+During the analysis, some files are automatically produced by the scripts and are saved on your working directory. These files are necessary for the subsequent steps in the analysis chain. It is furthermore mandatory that you always launch the scripts from the same working directory since the output files stored there can be correctly read as input files at the subsequent analysis steps.
```

Comment:
I think it would be good to be able to specify such a temporary directory. If those are e.g. log files, it is a good idea to keep them close to the place where the output files are.

Reply:
These files are: list of runs to be processed, list of source names, sh files and, for us (we heave to update the LST database) the txt files used to fill the nsb column. Log files are saved in a 'logs' folder in each output directory

Reply:
EDIT: now slurm output files are redirected to output folders with the logs. And maybe we could delete the sh files after they have been completed (with exit code 0) to 'clean' the working directory and leave only list of runs and sources there

Reply:
deleting the .sh files would be nice, but it is not easy, because they are the files that are actually executed by the cluster so you can remove them only after the job is done (and the script probably cannot remove itself, in particular if it is run in an array). In the tests that I did I also saw how many files are those (at least in DL1 step) and it is fine to clean them up by hand afterwards, so I close the comment